test/hotspot/jtreg/compiler/intrinsics/SortingDeoptimizationTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8318306
+ * @run main/othervm/timeout=200 -XX:+IgnoreUnrecognizedVMOptions -Xcomp -ea -esa -XX:CompileThreshold=100 -XX:+UnlockExperimentalVMOptions -server -XX:-TieredCompilation -XX:+DeoptimizeALot SortingDeoptimizationTest 1e-2 100 50
```

Comment:
Hi @vamsi-parasa, you are already passing -Xcomp which will trigger method compilation on first invocation, CompileThreshold may not be needed, You can pass -Xbatch instead to make compilation a blocking operation.

Reply:
But please verify that the issue still triggers. It may well be that -Xcomp is required to trigger one early allocation + deopt, followed by additional allocations once we reach the lowered threshold.

Reply:
> Hi @vamsi-parasa, you are already passing -Xcomp which will trigger method compilation on first invocation, CompileThreshold may not be needed, You can pass -Xbatch instead to make compilation a blocking operation.

Hi @jatin-bhateja , these flags are actually being used by the later tier tests and were given to me. Hence, they are being retained without any modifications.

Reply:
Ok, BTW I am able to reliably able to reproduce the issue with following JVM options.
 * @run main/othervm/timeout=200 -Xbatch -ea -esa -XX:-TieredCompilation -XX:+DeoptimizeALot SortingDeoptimizationTest .1 10 50000